crml/sylo/src/e2ee/benchmarking.rs

```diff
@@ -36,51 +82,25 @@ benchmarks! {
 		let p in 0 .. MAX_PKBS as u32;
 		let sender: T::AccountId = whitelisted_caller();
 		let device_id: DeviceId = 11;
-
-		let text_tuple0 = (Text::from(*b"t0m0"), Text::from(*b"t0m1"));
-		let text_tuple1 = (Text::from(*b"t1m0"), Text::from(*b"t1m1"));
-		let meta = Meta::from([text_tuple0, text_tuple1]);
-
-		let admin = Member::<T::AccountId>::new(
-			sender.clone(),
-			Vec::<MemberRoles>::from([MemberRoles::Admin, MemberRoles::Member]),
-			Meta::new()
-		);
-
-		let member = Member::<T::AccountId>::new(
-			account("member", 0, SEED),
-			Vec::<MemberRoles>::from([MemberRoles::Member]),
-			Meta::new()
-		);
-
-		let group_id_0 = T::Hashing::hash(b"group0");
-		let group_0 = Group::<T::AccountId, T::Hash>::new(
-			group_id_0,
-			Vec::<Member<T::AccountId>>::from([admin.clone(), member.clone()]),
-			Vec::new(),
-			meta.clone(),
-		);
-		<SyloGroups<T>>::insert(&sender, &group_id_0, group_0);
-
-		let group_id_1 = T::Hashing::hash(b"group1");
-		let group_1 = Group::<T::AccountId, T::Hash>::new(
-			group_id_1,
-			Vec::<Member<T::AccountId>>::from([admin, member]),
-			Vec::new(),
-			meta,
-		);
-		<SyloGroups<T>>::insert(&sender, &group_id_1, group_1);
-
-		let mut pre_key_bundles = Vec::<PreKeyBundle>::new();
-		for i in 0..p {
-			let mut pre_key_bundle = PreKeyBundle::from(*b"prekeybundle");
-			pre_key_bundle.extend_from_slice(&i.to_be_bytes());
-			pre_key_bundles.push(pre_key_bundle);
-		}
+		setup_groups::<T>(sender.clone());
+		let pre_key_bundles = create_pre_key_bundles::<T>(p);
 	}: _(RawOrigin::Signed(sender.clone()), device_id, pre_key_bundles)
 	verify {
 		assert!(<SyloDevice<T>>::devices(sender).contains(&device_id));
 	}
+
+	replenish_pkbs {
+		let p in 0 .. MAX_PKBS as u32;
+		let sender: T::AccountId = whitelisted_caller();
+		let device_id: DeviceId = 11;
+		setup_groups::<T>(sender.clone());
+		let pre_key_bundles = create_pre_key_bundles::<T>(0);
```

Comment:
Couples qs:
1) why declare `let pre_key_bundles = create_pre_key_bundles::<T>(0);` with 0?
2) also how does this allow two declarations of `let pre_key_bundles` seems like something idiosyncratic is happening? or either this doesn't run but the CI isn't checking benchmark features atm.
Would be good to comment what is happening here as it seems unusual

Reply:
The function replenish_pkbs extends the current pkbs for a user-device. In my setup, I need to first register the device with some or as it's the case here 0 pkbs and then benchmark replenish_pkbs extending it. As I wanted to allow the whole range of an extension for the worst-case scenario I used 0 as the initial number.

Reply:
I added a short comment there.